src/Microsoft.TestPlatform.Utilities/CodeCoverageRunSettingsProcessor.cs

```diff
@@ -0,0 +1,484 @@
+        /// </summary>
+        private class Exclusion
+        {
+            /// <summary>
+            /// Represents the <see cref="XPathNavigator"/> style path of the exclusion type.
+            /// </summary>
+            private string path;
+
+            /// <summary>
+            /// Gets the path components for the current exclusion type.
+            /// </summary>
+            public IEnumerable<string> PathComponents { get; }
+
+            /// <summary>
+            /// Gets the exclusion rules for the current exclusion type.
+            /// </summary>
+            public IDictionary<string, XmlNode> ExclusionRules { get; }
+
+            /// <summary>
+            /// Gets the actual exclusion type path generated from the individual path components.
+            /// </summary>
+            public string Path
+            {
+                get
+                {
+                    if (string.IsNullOrEmpty(this.path))
+                    {
+                        this.path = Exclusion.BuildPath(this.PathComponents);
+                    }
+
+                    return this.path;
+                }
+            }
+
+            /// <summary>
+            /// Constructs an <see cref="Exclusion"/> object.
+            /// </summary>
+            /// 
+            /// <param name="pathComponents">The path split in components.</param>
+            public Exclusion(IEnumerable<string> pathComponents)
+            {
+                this.path = string.Empty;
+                this.PathComponents = pathComponents;
+                this.ExclusionRules = new Dictionary<string, XmlNode>();
+            }
+
+            /// <summary>
+            /// Assembles a relative path from the path given as components.
+            /// </summary>
+            /// 
+            /// <returns>A relative path built from path components.</returns>
+            public static string BuildPath(IEnumerable<string> pathComponents)
+            {
+                var sb = new StringBuilder();
+                sb.Append(".");
+
+                foreach (var component in pathComponents)
+                {
+                    sb.Append("/");
+                    sb.Append(component);
+                }
+
+                return sb.ToString();
+            }
+        }
+        #endregion
+
+        #region Members
+        #region Default Settings String
+        /// <summary>
+        /// Represents the default settings for the code coverage data collector.
+        /// </summary>
+        private static readonly string DefaultSettings = string.Join(
+            Environment.NewLine,
+            @"<DataCollector uri=""datacollector://microsoft/CodeCoverage/2.0"" assemblyQualifiedName=""Microsoft.VisualStudio.Coverage.DynamicCoverageDataCollector, Microsoft.VisualStudio.TraceCollector, Version=16.0.0.0 " + @", Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"" friendlyName=""Code Coverage"">",
```

Comment:
As part of TraceDataCollector we have already file with default settings:

https://github.com/microsoft/vstest/blob/master/src/DataCollectors/TraceDataCollector/VanguardCollector/DefaultCodeCoverageConfig.xml

I would use this file to not have any discrepancies. Class CodeCoverageRunSettingsProcessor could be changed to get this default settings as argument in constructor.

Reply:
Done.